From af79db308c86d200d9fb76001167511408c68bce Mon Sep 17 00:00:00 2001
From: "Matthew J. Francis" <mjay.francis@gmail.com>
Date: Thu, 9 Oct 2014 13:02:18 +0800
Subject: fdo#84810 Avoid calc crash when using Paste Special - Link
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change-Id: I52c072578ccc0b9f75a850613056e74d400eb532
Reviewed-on: https://gerrit.libreoffice.org/11867
Reviewed-by: Kohei Yoshida <libreoffice@kohei.us>
Tested-by: Kohei Yoshida <libreoffice@kohei.us>
(cherry picked from commit 2b6619c597a791775e2d41a68f7e85ef75d1aaa2)
Reviewed-on: https://gerrit.libreoffice.org/11875
(cherry picked from commit b2707c0545d7cc762a9ccfbb8a3b3cea495195e2)
Reviewed-on: https://gerrit.libreoffice.org/11879
Reviewed-by: Caolán McNamara <caolanm@redhat.com>
Tested-by: Caolán McNamara <caolanm@redhat.com>

diff --git a/sc/source/core/data/document10.cxx b/sc/source/core/data/document10.cxx
index e468dd3..baf7314 100644
--- a/sc/source/core/data/document10.cxx
+++ b/sc/source/core/data/document10.cxx
@@ -76,6 +76,8 @@ bool ScDocument::CopyOneCellFromClip(
         return false;
 
     ScCellValue& rSrcCell = rCxt.getSingleCell();
+    const ScPatternAttr* pAttr = pClipDoc->GetPattern(aSrcPos);
+    rCxt.setSingleCellPattern(pAttr);
     if (rCxt.isAsLink())
     {
         ScSingleRefData aRef;
@@ -89,8 +91,6 @@ bool ScDocument::CopyOneCellFromClip(
     else
     {
         rSrcCell.set(pClipDoc->GetRefCellValue(aSrcPos));
-        const ScPatternAttr* pAttr = pClipDoc->GetPattern(aSrcPos);
-        rCxt.setSingleCellPattern(pAttr);
 
         // Check the paste flag to see whether we want to paste this cell.  If the
         // flag says we don't want to paste this cell, we'll return with true.
-- 
cgit v0.10.2

